File: javascript/triangle/triangle.js
export class Triangle{
  constructor(a,b,c){
    this.a = a
    this.b = b
    this.c = c
  }

  ORDER(){
    let x = []
      x.push(this.a, this.b, this.c)
    return x.sort((a,b) => b-a)   
  }

  ERROR(){
    return this.ORDER()[1] + this.ORDER()[2] < this.ORDER()[0] 
      || this.ORDER()[2] < 0 
      || this.ORDER()[2] === 0
  }

  kind(){
    if(this.ERROR()){
      throw new Error('error')
    }else{
      if(this.ORDER()[0] === this.ORDER()[2]) return 'equilateral'
      if(this.ORDER()[0] === this.ORDER()[1]) return 'isosceles'
      if(this.ORDER()[0] !== this.ORDER()[2]) return 'scalene'
    }
  }
}

File: ruby/triangle/triangle.rb
# frozen_string_literal: true

# Triangle exercism
class Triangle
  attr_reader :sides

  def initialize(sides)
    @sides = sides.drop_while(&:zero?)
  end

  def sides_order
    sides.sort
  end

  def invalid_triangle?
    return false if sides_order.empty?

    sides_order.first + sides_order[1] > sides_order.last
  end

  def equilateral?
    invalid_triangle? ? sides_order.first == sides_order.last : false
  end

  def isosceles?
    invalid_triangle? ? sides_order.reverse.first == sides_order.reverse[1] : false
  end

  def scalene?
    condition_a_and_b = sides_order.first != sides_order.last
    condition_b_and_c = sides_order[1] != sides_order.last
    invalid_triangle? ? condition_a_and_b && condition_b_and_c : false
  end
end
